Bundle/WidgetBundle/Form/WidgetStyleType.php

```diff
@@ -144,6 +150,76 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         });
     }
 
+    private function generateBackgroundFields(FormInterface $form, $type = null, $responsiveKey = false) {
```

Comment:
Please ease this heavy function which could be very simple:
$responsiveKey attr is required:
    - remove if ($responsiveKey !== false)
    - place $type = null after $responsiveKey in definition
replace bad switch with a nice if/else